src/core/core.scale.js

```diff
@@ -562,15 +561,7 @@ var Scale = Element.extend({
 	 * Handle margins and padding interactions
 	 * @private
 	 */
-	handleMargins: function() {
-		var me = this;
-		if (me.margins) {
-			me.paddingLeft = Math.max(me.paddingLeft - me.margins.left, 0);
-			me.paddingTop = Math.max(me.paddingTop - me.margins.top, 0);
-			me.paddingRight = Math.max(me.paddingRight - me.margins.right, 0);
-			me.paddingBottom = Math.max(me.paddingBottom - me.margins.bottom, 0);
-		}
-	},
+	handleMargins: helpers.noop,
```

Comment:
As `update` of a horizontal scale is no longer called more than once, `margins` doesn't reflect the changes in `padding{Top|Right|Bottom|Left}`. For better backward compatibility, what about the code below?

```js
handleMargins: function() {
	var me = this;
	if (me.margins) {
		me.margins.left = Math.max(me.paddingLeft, me.margins.left);
		me.margins.top = Math.max(me.paddingTop, me.margins.top);
		me.margins.right = Math.max(me.paddingRight, me.margins.right);
		me.margins.bottom = Math.max(me.paddingBottom, me.margins.bottom);
	}
},
```

Reply:
done